### DISCUSSION TITLE
Keybindings for window at position X

### DISCUSSION THREAD

**Discussion by RobLW** [paperwm/PaperWM#849] [Ideas]
Hi all,
On the back of [this suggestion](https://github.com/paperwm/PaperWM/discussions/842) of being able to load certain apps in the same window position, I thought it might also be handy to have the ability to map window positions to the keyboard as well.
Currently Super+Home and Super+End jump to the start and end of the window list which is really handy.

I'm thinking that since I don't use my function keys (F-keys top of keyboard) for anything other than controlling media it would be handy to map F1 to position-1, F2 to position-2 and F3 to position-3 etc etc. (or what ever the user prefers).

This is something I think I heard Primeagen (ThePrimeTime on youtube) mention and I came to understand how it's important to his workflow. Having certain apps which he can always access with the exact same key combo.

If it is possible to implement "App P always loads in Position X of current workspace", then we are able to map a key to jump to that position for fast navigation, F1 will take me to Position X where I always have App P loaded, maybe that's a web browser or your text editor/IDE of choice for example.

I imagine that the above is easier to implement rather than "mapping key-combo to jump to App P where ever it is in this workspace" as you might have multiple windows open for app P.

In terms of "always loading App P on position X", any additional windows will probably need to be loaded in the next "non-reserved" spot in the workspace maybe.

Hope the idea sounds sane, and interesting to implement (wish I knew how to code in JS). :)

Many thanks.

**Comment by jtaala** [comment]
Hey @RobLW, I quite like this idea (or mapping keys to positions in the tiling layout).

It would be a bit harder to implement the "App P always loads in Position X" - I wouldn't use that, but would be open to PRs for this (it could be part of our `winprops` system: https://github.com/paperwm/PaperWM?tab=readme-ov-file#setting-window-specific-properties).

In any case, I'll create an issue for the first part of this idea (`jump to position x in workspace`).

Cheers,

Jay.
